performance/scaling_data_ingestion.rst

```diff
@@ -3,79 +3,119 @@
+First, UPDATE statements cause bloat in the database and VACUUM needs to run regularly to clean up this bloat. In Citus, since VACUUM runs in parallel across worker nodes, your workloads are less likely to be impacted by VACUUM.
 
-Given that every command requires several round-trips to the workers, and no two commands can run on the same shard at the same time, update throughput is very low by default. However, if you know that the order of the queries doesn't matter (they are commutative), then you can turn on citus.all_modifications_commutative, in which case multiple commands can update the same shard concurrently.
+Second, these benchmark numbers show UPDATE throughput for standard Citus deployments. If you're on the Citus community edition, using statement-based replication, and you increased the default replication factor to 2, you're going to observe notably lower UPDATE throughputs. For this particular setting, Citus comes with additional configuration (citus.all_modifications_commutative) that may increase UPDATE ratios.
 
-For example, if your distributed table contains counters and all your DML queries are UPSERTs that add to the counters, then you can safely turn on citus.all_modifications_commutative since addition is commutative:
+Insert and Update: Throughput Checklist
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you're running the above pgbench benchmarks on a moderately sized Citus cluster, you can generally expect 10K-50K INSERTs per second. This translates to approximately 1 to 4 billion INSERTs per day. If you aren't observing these throughputs numbers, remember the following checklist:
+
+* Check the network latency between your application and your database. High latencies will impact your write throughput.
+* Ingest data using concurrent threads. If the roundtrip latency during an INSERT is 4ms, you can process 250 INSERTs/second over one thread. If you run 100 concurrent threads, you will see your write throughput increase with the number of threads.
+* Check whether the nodes in your cluster have CPU or disk bottlenecks. Ingested data passes through the coordinator node, so check whether your coordinator is bottlenecked on CPU.
+* Avoid closing connections between INSERT statements. This avoids the overhead of connection setup.
+* Remember that column size will affect insert speed. Rows with big JSON blobs will take longer than those with small columns like integers.
+
+Insert and Update: Latency
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The benefit of running INSERT, INSERT .. ON CONFLICT, UPDATE, or DELETE commands is that changes are immediately visible to other queries. When you issue an INSERT or UPDATE command, the Citus coordinator node routes this command to related worker node(s). The coordinator node also keeps connections to the workers open within the same session, which means subsequent commands will see lower response times.
```

Comment:
> The benefit of running INSERT, INSERT .. ON CONFLICT, UPDATE, or DELETE commands

The benefit as compared to what? Are there statements whose changes are not immediately visible?